File: src/color.rs
pub struct Color {
    r: u8,
    g: u8,
    b: u8,
}

impl Color {
    pub fn from_rgb(r: u8, g: u8, b: u8) -> Self {
        Self { r, g, b }
    }

    pub fn as_rgb_array(&self) -> [u8; 3] {
        [self.r, self.g, self.b]
    }
}

File: cpp/Color.cpp
#include "Color.h"

Color::Color(uint8_t r, uint8_t g, uint8_t b): r_{r}, g_{g}, b_{b} {
}

uint8_t Color::r() const {
    return r_;
}

uint8_t Color::g() const {
    return g_;
}

uint8_t Color::b() const {
    return b_;
}
